app/assets/javascripts/visualizations/highvis/map.coffee

```diff
@@ -399,127 +417,107 @@ $ ->
+          @configs.visibleLines = e.target.checked
           @delayedUpdate()
 
-        # Make heatmap select handler
-        ($ '#heatmapSelector').change (e) =>
-          element = e.target or e.srcElement
-          @configs.heatmapSelection = (Number element.value)
-
-          @delayedUpdate()
-
-        # Set up heatmap entry
-        ($ '#radius-text').keydown (e) =>
-          if e.which == 13
-            newRadius = Number ($ '#radius-text').val()
-            if isNaN newRadius
-              ($ '#radius-text').errorFlash()
-              return
-            # Guess new pixel radius
-            @heatmapPixelRadius = Math.ceil(@heatmapPixelRadius * newRadius / @configs.heatmapRadius)
-            @configs.heatmapRadius = newRadius
-            ($ '#heatmapSlider').slider "value", (Math.log @configs.heatmapRadius) / (Math.log 10)
-            @delayedUpdate()
+        $('#ckbx-cluster-markers').prop('checked', @configs.visibleClusters)
+        $('#ckbx-cluster-markers').click (e) =>
+          @configs.visibleClusters = e.target.checked
+          @start()
 
         # Set up heatmap slider
-        ($ '#heatmapSlider').slider
-          range: 'min'
-          value: (Math.log @configs.heatmapRadius) / (Math.log 10)
+        init =
+          value: Math.log(@configs.heatmapRadius) / Math.log(10)
           min: 0
           max: 6
-          values: 0
-          slide: (event, ui) =>
-            newRadius = Math.pow(10, Number ui.value)
-            # Guess new pixel radius
-            @heatmapPixelRadius = Math.ceil(@heatmapPixelRadius * newRadius / @configs.heatmapRadius)
-            @configs.heatmapRadius = newRadius
-            ($ '#radius-text').val("#{@configs.heatmapRadius}")
-            @delayedUpdate()
-
-        # Set up accordion
-        globals.configs.toolsOpen ?= 0
-
-        ($ '#toolControl').accordion
-          collapsible:true
-          active:globals.configs.toolsOpen
+        $('#heatmap-slider').attr(init)
+        $('#heatmap-slider').on 'input change', (e) =>
+          newRadius = Math.pow(10, Number(e.target.value))
+          # Guess new pixel radius
+          @heatmapPixelRadius =
+            Math.ceil(@heatmapPixelRadius * newRadius / @configs.heatmapRadius)
+          @configs.heatmapRadius = newRadius
+          $('#map-radius').val("#{@configs.heatmapRadius}")
+          @delayedUpdate()
 
-        ($ '#toolControl > h3').click ->
-          globals.configs.toolsOpen = (globals.configs.toolsOpen + 1) % 2
+        # Heatmap Radius Box
+        $('#map-radius').change (e) =>
+          newRadius = Number(e.target.value)
+          if isNaN(newRadius) then return $('#map-radius').errorFlash()
+
+          # Guess new pixel radius
+          @heatmapPixelRadius = Math.ceil(@heatmapPixelRadius * newRadius /
+            @configs.heatmapRadius)
+          @configs.heatmapRadius = newRadius
+          $('#heatmap-slider').val(Math.log(@configs.heatmapRadius) /
+            Math.log(10))
```

Comment:
Two problems with this:
1) `Math.log(10)` is a constant
2) Use the log division rule - `Math.log(@configs.heatmapRadius - 10)`